math/src/main/java/smile/math/random/MersenneTwister.java

```diff
@@ -75,7 +75,7 @@ public MersenneTwister(int seed) {
 
     @Override
     public void setSeed(long seed) {
-        setSeed(seed % UniversalGenerator.BIG_PRIME);
+        setSeed((int)seed % UniversalGenerator.BIG_PRIME);

```

Comment:
It should be
```
setSeed((int) (seed % UniversalGenerator.BIG_PRIME));
```
Otherwise, we will lose useful digits in the `seed`. Thanks.

Reply:
Good catch. I fixed it in new commit